### PR TITLE
Stricter infe parsing

### PR DESCRIPTION
**Pull request by y-guyon**

### CODE REVIEW

**Review by vrabaud — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -94,6 +94,8 @@ The changes are relative to the previous release, unless the baseline is specifi
 * Fix 'iloc' box parsing bugs that may have wrongly accepted, rejected or parsed
   some files with rare values of offset_size, length_size, base_offset_size and
   index_size.
+* 'infe' boxes with an item_type different of 'mime' and without a
```
Comment (wantehchang) [inline CHANGELOG.md:97]: Nit: Change "different of" to "different from"

File: src/read.c [hunk 2]
```diff
@@ -2670,18 +2671,21 @@ static avifResult avifParseItemInfoEntry(avifMeta * meta, const uint8_t * raw, s
...
-        AVIF_CHECKERR(avifROStreamReadString(&s, NULL, 0), AVIF_RESULT_BMFF_PARSE_FAILED); // string item_name; (skipped)
-        AVIF_CHECKERR(avifROStreamReadString(&s, contentType.contentType, CONTENTTYPE_SIZE), AVIF_RESULT_BMFF_PARSE_FAILED); // string content_type;
+        AVIF_CHECKERR(avifROStreamReadString(&s, contentType.contentType, CONTENTTYPE_SIZE), AVIF_RESULT_BMFF_PARSE_FAILED); // utf8string content_type;
+        // utf8string content_encoding; //optional
     } else {
         memset(&contentType, 0, sizeof(contentType));
     }
+    // if (item_type == 'uri ') {
+    //  utf8string item_uri_type;
+    // }
```
Comment (wantehchang) [inline src/read.c:2688]: Nit/optional: This comment (lines 2686-2688) probably should go into the `else` block at line 2684.

File: src/write.c [hunk 3]
```diff
@@ -2600,10 +2601,14 @@ avifResult avifEncoderFinish(avifEncoder * encoder, avifRWData * output)
...
-        AVIF_CHECKRES(avifRWStreamWriteChars(&s, item->infeName, item->infeNameSize)); // string item_name; (writing null terminator)
-        if (item->infeContentType && item->infeContentTypeSize) { // string content_type; (writing null terminator)
-            AVIF_CHECKRES(avifRWStreamWriteChars(&s, item->infeContentType, item->infeContentTypeSize));
-        }
+        AVIF_CHECKRES(avifRWStreamWriteChars(&s, item->infeName, item->infeNameSize)); // utf8string item_name; (writing null terminator)
+        if (!memcmp(item->type, "mime", 4)) {
+            AVIF_CHECKRES(avifRWStreamWriteChars(&s, item->infeContentType, item->infeContentTypeSize)); // utf8string content_type; (writing null terminator)
+            // utf8string content_encoding; //optional
+        }
+        // if (item_type == 'uri ') {
```
Comment (wantehchang) [inline src/write.c:2609]: Nit: Add "else" before the "if" in the comment.

But it seems that we should return `AVIF_RESULT_NOT_IMPLEMENTED` if `item->type` is equal to `"uri "`.

**Review by wantehchang — APPROVED** [review]

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**